app/src/main/java/app/accrescent/client/ui/SettingsScreen.kt

```diff
@@ -67,27 +67,35 @@ fun SettingsScreen(modifier: Modifier = Modifier, viewModel: SettingsViewModel =
     ) {
         if (context.isPrivileged()) {
             SettingGroupLabel(stringResource(R.string.app_updates), Modifier.padding(top = 16.dp))
+            val setRequireUserAction: (Boolean) -> Unit = {
+                coroutineScope.launch { viewModel.setRequireUserAction(it) }
+            }
             Setting(
                 label = stringResource(R.string.require_user_action),
                 description = stringResource(R.string.require_user_action_desc),
-                modifier = Modifier.fillMaxWidth(),
+                modifier = Modifier.fillMaxWidth()
+                    .clickable { setRequireUserAction(!dynamicColor) },
             ) {
                 Switch(
                     checked = requireUserAction,
-                    onCheckedChange = { coroutineScope.launch { viewModel.setRequireUserAction(it) } },
+                    onCheckedChange = { setRequireUserAction(it) },
```

Comment:
I think `onCheckedChange` can be set to `null` in the switches since the surrounding `Setting` touch target covers that of the switch. I believe the same applies to the `IconButton`.

Reply:
Nope, I tested that already. They didn't work in that area when I did that.

Reply:
To clarify about `IconButton`, I think it can be removed entirely in favor of the contained `Icon` since there's no need for it to be a button if the encompassing `Setting` is clickable.

Reply:
Oh yeah. That works.

Reply:
> Nope, I tested that already. They didn't work in that area when I did that.

Are you sure you didn't do something slightly different? I just tested both in emulator and on a physical device and it seems to work.

Reply:
I think I only tested the iconbutton, not the switch. So the switches toggle even when tapping on them directly?

Reply:
Yes. From my testing, tapping on the switches directly works even when `onCheckedChange` is `null`.

Reply:
OK, I changed those to null and also removed the values I added since its only used once so there's no point in it